File: src/test/scala/com/github/kayjamlang/tests/expressions/value/ValueExpressionIntegerTest.scala
package com.github.kayjamlang.tests.expressions.value

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.expressions.ValueExpression
import org.junit.Assert._
import org.junit.{BeforeClass, Test}


object ValueExpressionIntegerTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("12345"))
  }
}

class ValueExpressionIntegerTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = ValueExpressionIntegerTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[ValueExpression], expression getClass)
    val constant = expression.asInstanceOf[ValueExpression]
    assertEquals(12345, constant.value)
  }
}

File: src/test/java/com/github/kayjamlang/tests/expressions/value/ValueExpressionIntegerTest.java
package com.github.kayjamlang.tests.expressions.value;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.expressions.ValueExpression;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class ValueExpressionIntegerTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("12345"));
    }

    @Test
    public void test() throws Exception {
        Expression expression = parser.readExpression();

        assertNotNull(expression);
        assertSame(ValueExpression.class, expression.getClass());

        ValueExpression constant = (ValueExpression) expression;
        assertEquals(12345, constant.value);
    }
}